datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/build_traps_db.py

```diff
@@ -0,0 +1,294 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import json
+import os
+from contextlib import ExitStack
+from functools import lru_cache
+
+import click
+import yaml
+from pysmi import error
+from pysmi.codegen import JsonCodeGen
+from pysmi.compiler import MibCompiler
+from pysmi.parser import SmiV1CompatParser
+from pysmi.reader import getReadersFromUrls
+from pysmi.searcher import AnyFileSearcher
+from pysmi.writer import FileWriter
+
+from datadog_checks.dev import TempDir
+
+from ...console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success, echo_warning
+from .constants import MIB_SOURCE_URL
+
+# Unique identifiers of traps in json-compiled MIB files.
+NOTIFICATION_TYPE = 'notificationtype'
+
+
+@click.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Generate a traps database that can be used by the '
+    'Datadog Agent for resolving Traps OIDs to readable strings.',
+)
+@click.option(
+    '--mib-sources',
+    '-s',
+    help='Url or a path to a directory containing the dependencies for [mib_files...].'
+    'Traps defined in these files are ignored.',
+)
+@click.option(
+    '--compiled-mibs-sources',
+    '-c',
+    help='Path to a directory where to store the intermediate compiled MIBs in JSON format. Defaults to a temp dir',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    help='Path to a directory where to store the created traps database file per MIB.'
+    'Recommended option, do not use with --output-file',
+)
+@click.option(
+    '--output-file',
+    help='Path to a file to store a compacted version of the traps database file. Do not use with --output-dir',
+)
+@click.option('--use-json', help='Use json instead of yaml for the output file(s).', is_flag=True)
+@click.argument(
+    'mib-files',
+    nargs=-1,
+)
+def build_traps_db(mib_sources, compiled_mibs_sources, output_dir, output_file, use_json, mib_files):
+    """Builds yaml formatted documents containing various information about traps. These file scan be used by
+    the Datadog Agent to enrich trap data.
+    This command is intended for "Network Devices Monitoring" users who need to enrich traps that are not automatically
+    supported by Datadog.
+
+    The expected workflow is as such:\n
+    1- Identify a type of device that is sending traps that Datadog does not already recognize.\n
+    2- Fetch all the MIBs that we do not support.\n
+    3- Run `ddev meta snmp build-traps-db -o ./output_dir/ /path/to/my/mibs/*`\n
+    """
+
+    # Defaulting to github.com/DataDog/mibs.snmplabs.com/
+    mib_sources = [mib_sources] if mib_sources else [MIB_SOURCE_URL]
+
+    if output_file:
+        allowed_extensions = ['.json'] if use_json else ['.yml', '.yml']
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
I would go as far as removing the extension, if available. Given `--use-json` option, user might end up generating `yaml` into ` json` extended file, or viceversa. I'd reduce logic here and have extension only based on `--use-json`

Reply:
Replacing the existing extension with the correct one instead of simply adding it ✅